crates/typst/src/visualize/paint.rs

```diff
@@ -1,48 +1,59 @@
-use std::fmt::{self, Debug, Formatter};
-
 use ecow::EcoString;
 
-use crate::foundations::{cast, Repr};
-use crate::visualize::{Color, Gradient, GradientRelative};
+use crate::foundations::{cast, Repr, Smart};
+use crate::visualize::{Color, Gradient, RelativeTo};
+
+use super::Pattern;
 
 /// How a fill or stroke should be painted.
-#[derive(Clone, Eq, PartialEq, Hash)]
+#[derive(Debug, Clone, PartialEq, Hash)]
 pub enum Paint {
     /// A solid color.
     Solid(Color),
     /// A gradient.
     Gradient(Gradient),
+    /// A pattern.
+    Pattern(Pattern),
 }
 
 impl Paint {
     /// Unwraps a solid color used for text rendering.
     pub fn unwrap_solid(&self) -> Color {
         match self {
             Self::Solid(color) => *color,
-            Self::Gradient(_) => panic!("expected solid color"),
+            Self::Gradient(_) | Self::Pattern(_) => panic!("expected solid color"),
+        }
+    }
+
+    /// Gets the relative coordinate system for this paint.
+    pub fn relative(&self) -> Smart<RelativeTo> {
+        match self {
+            Self::Solid(_) => Smart::Auto,
+            Self::Gradient(gradient) => gradient.relative(),
+            Self::Pattern(pattern) => pattern.relative,
         }
     }
 
     /// Turns this paint into a paint for a text decoration.
     ///
     /// If this paint is a gradient, it will be converted to a gradient with
-    /// relative set to [`GradientRelative::Parent`].
+    /// relative set to [`RelativeTo::Parent`].
     pub fn as_decoration(&self) -> Self {
         match self {
             Self::Solid(color) => Self::Solid(*color),
             Self::Gradient(gradient) => {
-                Self::Gradient(gradient.clone().with_relative(GradientRelative::Parent))
+                Self::Gradient(gradient.clone().with_relative(RelativeTo::Parent))
+            }
+            Self::Pattern(pattern) => {
+                Self::Pattern(pattern.clone().with_relative(RelativeTo::Parent))
             }
         }
     }
 }
 
-impl Debug for Paint {
```

Comment:
Why did you remove this `Debug` impl? It was intentional to reduce the noise when printing.

Reply:
My bad, I thought it was some code from a previous gradient PR that I had forgotten to remove, it's back in now